google/api/http.proto

```diff
@@ -346,6 +339,18 @@ message HttpRule {
   // message type.
   string body = 7;
 
+  // The name of fields whose values are mapped to HTTP headers in a request. If
+  // a header has a case-insensitive match to a key in this map, the field in
```

Comment:
What happens if multiple map values refer to the same field?  Would it make more sense to invert the mapping so that request fields rather than headers benefit from the uniqueness of keys?

I can't think of a realistic use case for mapping the same header to multiple fields, but it still seems better semantically than mapping multiple headers to the same field (which might even have undefined behavior if not otherwise specified here, e.g. given a naive implementation that effectively depends on the (undefined) ordering of map entries).

Reply:
I don't think either is realistic - both keys and values should just be unique, so I think i'll just update it to state as such?

I can't think of a good reason to allow either multiple headers mapping to the same field, or multiple fields mapping to the same header.

Reply:
updated to make them all unique, and we can carry on the larger discussion in the other thread.